app/src/androidTestE2eTest/java/io/appium/uiautomator2/unittest/test/DeviceCommandsTest.java

```diff
@@ -0,0 +1,449 @@
+        response = getAttribute(response.getElementId(), "text");
+        assertEquals("Content", response.getValue());
+    }
+
+    /**
+     * test to find element using "-android automator" property
+     */
+    @Test
+    public void findElementUsingUiAutomatorTest() throws JSONException {
+        scrollTo("Views"); // Due to 'Views' option not visible on small screen
+        Response response = findElement(By.accessibilityId("Views"));
+        clickAndWaitForStaleness(response.getElementId());
+
+        By androidUiAutomator = By.androidUiAutomator("new UiScrollable(new UiSelector()"
+                + ".resourceId(\"android:id/list\")).scrollIntoView("
+                + "new UiSelector().text(\"Radio Group\"));");
+        response = findElement(androidUiAutomator);
+        assertTrue(androidUiAutomator + " should be found", response.isSuccessful());
+        click(response.getElementId());
+
+        response = waitForElementInvisibility(response.getElementId());
+        assertFalse(androidUiAutomator + " should not be found", response
+                .isSuccessful());
+    }
+
+    /**
+     * test to find elements using "-android automator" property
+     */
+    @Test
+    public void findElementsUsingUiAutomatorTest() throws JSONException {
+        scrollTo("Views"); // Due to 'Views' option not visible on small screen
+        Response response = findElement(By.accessibilityId("Views"));
+        clickAndWaitForStaleness(response.getElementId());
+
+        By by = By.androidUiAutomator("resourceId(\"android:id/text1\")");
+        response = findElements(by);
+        assertTrue(by + " should be found", response.isSuccessful());
+
+        JSONArray elements = response.getValue();
+        int elementCount = getJsonObjectCountInJsonArray(elements);
+        assertTrue("Elements Count in views screen should at least > 5, " +
+                "in all variants of screen sizes, but actual: " + elementCount, elementCount > 5);
+    }
+
+    /**
+     * Test for findElements
+     */
+    @Test
+    public void findElementsTest() {
+        By by = By.className("android.widget.TextView");
+        Response response = findElements(by);
+        assertTrue(by + " should be found", response.isSuccessful());
+        JSONArray elements = response.getValue();
+        int elementCount = getJsonObjectCountInJsonArray(elements);
+        assertTrue("Elements Count in Home launch screen should at least > 5, " +
+                "in all variants of screen sizes, but actual: " + elementCount, elementCount > 5);
+    }
+
+    /**
+     * Test for Device size
+     *
+     * @throws JSONException
+     */
+    @Test
+    public void getDeviceSizeTest() throws JSONException {
+        JSONObject value = getDeviceSize().getValue();
+        Integer height = value.getInt("height");
+        Integer width = value.getInt("width");
+        assertTrue("device window height is zero(0), which is not expected", height > 479);
```

Comment:
Have you ever tried hamcrest's  `assertThat`?

Reply:
I think it is outside of this pr - most of existing tests should be optimized.